src/controllers/CategoryController.ts

```diff
@@ -0,0 +1,31 @@
+import { CategoryModel } from "../models/Category";
+import asyncHandler from "express-async-handler";
+
+interface CategoryController {
+    getCategories?: any;
+}
+
+const categoryController: CategoryController = {};
+
+const getCategories = asyncHandler(async (req, res) => {
+    let document;
+    if (req.query.name) {
+        document = await CategoryModel.findOne({ "name": req.query.name });
+        if (!document) {
+            res.status(404);
+            throw new Error("Category not found");
```

Comment:
Does this lead the backend to crash or does it "log" an error?

Reply:
It doesn't crash, it displays the error in the log and a stack trace